### PR TITLE
Canvas Event Handlers Added

### PR DESCRIPTION
**Pull request by arshithsuresh**

### CODE REVIEW

File: src/app/core/logging/eventInspectorLogger.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    eventFired:Subject<string> = new Subject();
+    eventFired$ = this.eventFired.asObservable();
+
+    constructor(){
+    }
+
+    logEvent(event:ICanvasEvent): void {
+        
+    }
+}
```
Comment (isham-mohamed) [inline src/app/core/logging/eventInspectorLogger.ts]: remove unused methods

File: src/app/core/events/objectRotatedEvent.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        this.active=true;
+    }
+
+    disableEvent(): void {
+        this.active=false;
+    }
+
+    getEventName():string{
+        return this.eventName;
+    }
```
Comment (isham-mohamed) [inline src/app/core/events/objectRotatedEvent.ts]: can be moved to abstract class

File: src/app/core/events/objectMovedEvent.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { ICanvasEvent } from "./canvasEvent";
+
+export class ObjectMovedEvent implements ICanvasEvent{
+
+    active: boolean=true;
+    eventName: string = "object:moving";
+    eventMessage:string = "Object Moved";
+
```
Comment (isham-mohamed) [inline src/app/core/events/objectMovedEvent.ts]: check with moved event

File: src/app/core/logging/event-handle.service.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    initEventHandlers(canvas:fabric.Canvas){
+        this.registedEvents.forEach(event=>{
+            canvas.on(event.getEventName()
+            ,(e)=>{
+                if(event.active)
+                {
+                    let eventMessage = event.getEventMessage(e);
+                    this.eventInspectorService.eventFired.next(eventMessage);
+                }                
+            })
```
Comment (isham-mohamed) [inline src/app/core/logging/event-handle.service.ts]: move to canvas component

**Review by isham-mohamed — CHANGES_REQUESTED** [review]

File: src/app/core/logging/event-handle.service.ts (hunk 4 above)
Comment (arshithsuresh) [reply src/app/core/logging/event-handle.service.ts]: Moved

File: src/app/core/events/objectRotatedEvent.ts (hunk 2 above)
Comment (arshithsuresh) [reply src/app/core/events/objectRotatedEvent.ts]: Done

File: src/app/core/logging/eventInspectorLogger.ts (hunk 1 above)
Comment (arshithsuresh) [reply src/app/core/logging/eventInspectorLogger.ts]: Removed

**Review by isham-mohamed — APPROVED** [review]

**Pull request merged**